src/v/cluster/archival_metadata_stm.cc

```diff
@@ -192,7 +175,106 @@ ss::future<std::error_code> archival_metadata_stm::do_add_segments(
         co_return errc::replication_error;
     }
 
-    for (const auto& segment : segments) {
+    co_return errc::success;
+}
+
+ss::future<std::error_code> archival_metadata_stm::do_truncate(
+  model::offset start_rp_offset,
+  ss::lowres_clock::time_point deadline,
+  std::optional<std::reference_wrapper<ss::abort_source>> as) {
+    {
+        auto now = ss::lowres_clock::now();
+        auto timeout = now < deadline ? deadline - now : 0ms;
+        if (!co_await sync(timeout)) {
+            co_return errc::timeout;
+        }
+    }
+
+    if (as) {
+        as->get().check();
+    }
+
+    auto so = _manifest.get_start_offset();
+    if (so && so.value() > start_rp_offset) {
+        co_return errc::success;
+    }
+
+    storage::record_batch_builder b(
+      model::record_batch_type::archival_metadata, model::offset(0));
+    iobuf key_buf = serde::to_iobuf(truncate_cmd::key);
+    auto record_val = truncate_cmd::value{.start_offset = start_rp_offset};
+    iobuf val_buf = serde::to_iobuf(record_val);
+    b.add_raw_kv(std::move(key_buf), std::move(val_buf));
+
+    auto batch = std::move(b).build();
+
+    auto ec = co_await do_replicate_commands(std::move(batch), deadline, as);
+    if (!ec) {
+        co_return ec;
+    }
+
+    vlog(
+      _logger.info,
+      "truncate command replicated, truncated up to {}, remote start_offset: "
+      "{}, last_offset: {}",
+      start_rp_offset,
+      _start_offset,
+      _last_offset);
+
+    co_return errc::success;
+}
+
+// todo: return result
```

Comment:
nit: is this comment still valid ?

Reply:
removed that, I can't even remember what does this mean